dali_tf_plugin/dali_tf_plugin_utils.py

```diff
@@ -0,0 +1,150 @@
+def get_tf_compiler_version():
+    tensorflow_path = get_module_path('tensorflow')
+    tensorflow_libs = find('libtensorflow_framework*so*', tensorflow_path)
+    if not tensorflow_libs:
+        return ''
+    lib = tensorflow_libs[0]
+    cmd = 'strings -a ' + lib + ' | grep "GCC: ("'
+    s = str(subprocess.check_output(cmd, shell=True))
+    version = re.search("GCC:\s*\(.*\)\s*(\d+.\d+).\d+", s).group(1)
+    return version
+
+# Get current tensorflow version
+def get_tf_version():
+    try:
+        import pkg_resources
+        s = pkg_resources.get_distribution("tensorflow-gpu").version
+    except:
+        # pkg_resources.get_distribution doesn't work well with conda installed packages
+        try:
+            import tensorflow as tf
+            s = tf.__version__
+        except:
+            return ""
+    version = re.search("(\d+.\d+).\d+", s).group(1)
+    return version
+
+# Get C++ compiler
+def get_cpp_compiler():
+    return os.environ.get('CXX') or 'g++'
+
+# Get C++ compiler version
+def get_cpp_compiler_version():
+    cmd = get_cpp_compiler() + ' --version | head -1 | grep "[c|g]++ ("'
+    s = str(subprocess.check_output(cmd, shell=True).strip())
+    version = re.search("[g|c]\+\+\s*\(.*\)\s*(\d+.\d+).\d+", s).group(1)
+    return version
+
+# Runs `which` program
+def which(program):
+    try:
+        return subprocess.check_output('which ' + program, shell=True).strip()
+    except:
+        return None
+
+# Checks whether we are inside a conda env
+def is_conda_env():
+    return "" != os.environ.get('CONDA_PREFIX')
+    
+# Get compile and link flags for installed tensorflow
+def get_tf_build_flags():
+    tf_cflags = ''
+    tf_lflags = ''
+    try:
+        import tensorflow as tensorflow
+        tf_cflags=" ".join(tensorflow.sysconfig.get_compile_flags())
+        tf_lflags=" ".join(tensorflow.sysconfig.get_link_flags())
+    except:
+        tensorflow_path = get_module_path('tensorflow')
+        if tensorflow_path is not '':
+            tf_cflags=" ".join(["-I" + tensorflow_path + "/include",  "-I" + tensorflow_path + "/include/external/nsync/public", "-D_GLIBCXX_USE_CXX11_ABI=0"])
+            tf_lflags=" ".join(["-L" + tensorflow_path, "-ltensorflow_framework"])
+
+    if tf_cflags is '' and tf_lflags is '':
+        raise ImportError('Could not find Tensorflow. Tensorflow must be installed before installing nvidia-dali-tf-plugin')
+    return (tf_cflags, tf_lflags)
+
+# Get compile and link flags for installed DALI
+def get_dali_build_flags():
+    dali_cflags = ''
+    dali_lflags = ''
+    try:
+        import nvidia.dali.sysconfig as dali_sc
+        dali_lib_path = dali_sc.get_lib_dir()
+        dali_cflags=" ".join(dali_sc.get_compile_flags())
+        dali_lflags=" ".join(dali_sc.get_link_flags())
+    except:
+        dali_path = get_module_path('nvidia/dali')
+        if dali_path is not '':
+            dali_cflags=" ".join(["-I" + dali_path + "/include", "-D_GLIBCXX_USE_CXX11_ABI=0"])
```

Comment:
Actually, we could have 3 sets of flags here:
C flags
C++ flags
Link flags
For TF plugin we use DALI C API and don't have to use C++ flags which might conflict with TF C++ flags and make compiler raise warnings.

Reply:
:+1 (task DALI-1063 created)